src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java

```diff
@@ -547,23 +547,43 @@ public WriteLock appendLock() {
     @NotNull
     @Override
     public ExcerptTailer createTailer(String id) {
-        throwExceptionIfClosed();
+        verifyTailerPreconditions(id);
 
         LongValue index = id == null
                 ? null
                 : indexForId(id);
-        final StoreTailer storeTailer = new StoreTailer(this, pool, index);
+        LongValue indexVersion = id == null
+                ? null
+                : indexVersionForId(id);
+
+        final StoreTailer storeTailer = new StoreTailer(this, pool, index, indexVersion);
         directoryListing.refresh(true);
         storeTailer.singleThreadedCheckReset();
         return storeTailer;
     }
 
+    private void verifyTailerPreconditions(String id) {
+        // Preconditions for all tailer types
+        throwExceptionIfClosed();
+
+        // Named tailer preconditions
+        if (id == null) return;
+        if (appendLock.locked()) {
+            throw new NamedTailerNotAvailableException(id, NamedTailerNotAvailableException.Reason.NOT_AVAILABLE_ON_SINK);
+        }
```

Comment:
Should this be more specific and check that the tailer is not a replicated named tailer. I don't think we want to remove the ability to use non-replicated named tailers on sinks.

I wonder if we could designate that replicated named tailers start with a prefix or something? `replicated:somename` instead of having to specify them in a config somewhere.

Reply:
Hi @nicktindall good spot. You are right - this needs to change. I will follow the `replicated:somename` approach and update the PR

Reply:
This has now been updated